chunky/src/java/se/llbit/chunky/resources/Texture.java

```diff
@@ -1033,7 +1035,10 @@ public void getColor(Ray ray) {
    * @return color
    */
   public final float[] getColor(int x, int y) {
-    return linear[width * y + x];
+    // TODO Not allocate here
```

Comment:
I think all other `getColor` methods just assume that the returned array is never altered and don't allocate a new array.

Reply:
I'm not sure i get what you are saying, can you elaborate?
All the other `getColor` methods end up calling this one. I made that comment thinking that allocating a new array isn't good on performances, a better way would likely be to pass an array as an output parameter but that would need changing every model class (and more)

Reply:
It didn't allocate a new array before, so it probably doesn't need to now.

Reply:
Oh ok. The problem is caller expect an array with 4 elements containing r,g,b,a at indices 0,1,2,3 but now the component they want isn't in another index so i can't give them the array. If there was a way to give them as array object that acts like a view of some bigger array without allocating that would do the job but i don't think it exists

Reply:
Even then, allocating that object would probably be more expensive than allocating an array of 4 floats. It's fine, I'll merge it. :+1:

Reply:
That's true